src/wasm/wasm-binary.cpp

```diff
@@ -636,8 +636,13 @@ void WasmBinaryWriter::visitSetLocal(SetLocal *curr) {
 }
 
 void WasmBinaryWriter::visitGetGlobal(GetGlobal *curr) {
-  if (debug) std::cerr << "zz node: GetGlobal " << (o.size() + 1) << std::endl;
-  o << int8_t(BinaryConsts::GetGlobal) << U32LEB(getGlobalIndex(curr->name));
+  if (curr->name.is()) {
```

Comment:
I don't know, but here in findImports:
```
  void findImports() {
    findImportsByBase(wasm, MEMORY_BASE, [&](Name name) {
      memoryBaseGlobals.insert(name);
    });
    if (memoryBaseGlobals.size() == 0) {
      Fatal() << "no memory base was imported";
    }
    findImportsByBase(wasm, TABLE_BASE, [&](Name name) {
      tableBaseGlobals.insert(name);
    });
    if (tableBaseGlobals.size() == 0) {
      Fatal() << "no table base was imported";
    }
  }
```
As you can see in the code above, it assumes memoryBaseGlobals and tableBaseGlobals exist and `Fatal()`'s if not. In my change I removed those two conditional calls to `Fatal()` and "fixed" a couple uses via name.is() tests. So my guess is that @kripken added the test in findImports because there is an assumption that they "always" exist somewhere.

I'll try to find more information.